backend/core/src/listings/listings.service.ts

```diff
@@ -212,6 +216,119 @@ export class ListingsService {
     return listing.jurisdiction.id
   }
 
+  public async getApprovingUserEmails(): Promise<string[]> {
+    const approvingUsers = await this.userRepository
+      .createQueryBuilder("user")
+      .select(["user.email"])
+      .leftJoin("user.roles", "userRoles")
+      .where("userRoles.is_admin = :is_admin", {
+        is_admin: true,
+      })
+      .getMany()
+    const approvingUserEmails: string[] = []
+    approvingUsers?.forEach((user) => user?.email && approvingUserEmails.push(user.email))
+    return approvingUserEmails
+  }
+
+  public async getNonApprovingUserInfo(
+    listingId: string,
+    jurisId: string,
+    getPublicUrl = false
+  ): Promise<{ emails: string[]; publicUrl?: string | null }> {
+    const selectFields = ["user.email", "jurisdictions.id"]
+    getPublicUrl && selectFields.push("jurisdictions.publicUrl")
+    const nonApprovingUsers = await this.userRepository
+      .createQueryBuilder("user")
+      .select(selectFields)
+      .leftJoin("user.leasingAgentInListings", "leasingAgentInListings")
+      .leftJoin("user.roles", "userRoles")
+      .leftJoin("user.jurisdictions", "jurisdictions")
+      .where(
+        new Brackets((qb) => {
+          qb.where("userRoles.is_partner = :is_partner", {
+            is_partner: true,
+          }).andWhere("leasingAgentInListings.id = :listingId", {
+            listingId: listingId,
+          })
+        })
+      )
+      .orWhere(
+        new Brackets((qb) => {
+          qb.where("userRoles.is_jurisdictional_admin = :is_jurisdictional_admin", {
```

Comment:
This makes me think we might want this flag on the backend so that we don't have to make a code change in Alameda to change the permission level (future consideration)

Reply:
Good point! Will keep that in mind when we bring it over 📃

Reply:
I agree, I think adding an environment variable to determine that would be ideal so that the code is the same everywhere and configurable. 

We also might want to have that variable as a variable on the jurisdiction table (or the future feature flag table), that way it can be different depending on the jurisdiction. e.g Alameda goes to super admin, but SMC goes to jurisdictional admin.

That can be done as a separate ticket though